### ISSUE TITLE
Update Kind registry configuration

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes-sigs/cluster-api#8736]
### What would you like to be added (User Story)?

As a developer I would like to have a simpler way of setting up a local registry for testing.

### Detailed Description

Since containerd deprecated CRI hosts config Kind has to use a more complex method for setting up the local registry. We kept the CAPI `kind-install-for-capd.sh` script [the same as Kind's version upstream](https://github.com/kubernetes-sigs/cluster-api/pull/8731), but we should fix this as soon as it's fixed in Kind.

Reference kind issue: https://github.com/kubernetes-sigs/kind/issues/2875

### Label(s) to be applied

/kind feature
One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels.

**Comment by killianmuldoon** [comment]
/assign

I can track this

**Comment by fabriziopandini** [comment]
@killianmuldoon is this fixed now that the PR is merged?

**Comment by killianmuldoon** [comment]
>is this fixed now that the PR is merged

No, this is to track upcoming changes in kind so we improve / modify this as they change it upstream.
